Moved filesystem/remote ca-store management to separate package

pdspicer opened the pull request:
Disclaimer: please feel free to completely ignore this PR if you want, it has breaking changes and actually moves the bulk of the code to a separate project I wrote based on ssl-root-cas to achieve some extra functionality.

I moved the entirety of the pulls from mozilla along with the parsing and storage of .pems to a separate project, which I've modified to include some additional functionality including the ability to load directories of .pem files in addition to loading from .js files (as this module had already been doing). You and Forrest are both mentioned as the primary authors of the code off which it is based (I'm happy to get rid of this if you'd prefer not to be associated), as I wanted to make sure you were still being credited for your work.

I wanted to put in this PR out of consideration, just in case you found it to be of interest. I realize this is a massive change and doesn't really accomplish much for the ssl-root-cas project on the whole, other than perhaps enabling some of the capabilities vis-a-vis .pem loading that were discussed in the issues a ways back. Once again please feel free to ignore it completely.

Thanks for all your work on this project, its helped me out a lot in the past and given me some good direction for my current work.

coolaj86 commented:
I'm all about progress.

You say there are breaking changes? Would you mind updating the documentation accordingly and bumping the major version?

If so I'll add you as a maintainer and give you push and publish access here and on npm and trust you to test your changes before publishing.

:)

pdspicer commented:
Absolutely! I'll get to that later today.

One of the changes I wanted to get your thoughts on actually, as I can think of reasons for doing it the way you did: I switched the prepublish script to a postinstall script, that way updating the latest scripts doesn't require a version change, but I could see how this might be an issue for installation on a locked down system that for whatever reason can't access the mozilla site. Was this similar to your line of thinking? I suppose it could be done both ways as long as postinstall always returns a 0 exit code so that it doesn't interfere with the installation process.

coolaj86 commented:
It was originally a postinstall script, but a number of people had issues due to various weird reasons.

Also, it's good to know that the exact same version will work exactly the same for everyone, hence the `latest` option.

pdspicer commented:
Updated README and added changelog to outline major changes.

pdspicer commented:
Hi @coolaj86 let me know if/when you have an opportunity to look over the latest, I didn't bump version yet but everything is reflected in README and changelog, also reverted the npm job to prepublish

coolaj86 commented:
What's the reason to nix `addFile()`? And if there's a particularly good reason, could we reasonably maintain it for backwards compat with a deprecation notice?

Also, I just added in the `create()` because I found that I don't like singletons. They always tend to cause trouble down the road. What's the reason that you removed it?

These are just questions. You're the one doing the work to move this forward so I give you equal vote on how it moves forward with v2.x.

pdspicer commented:
I don't believe I removed `create()`... if you are seeing differently somewhere let me know where and I'll fix it but it looks to me like its still there. 
As for `addFile()`, happy supporting with a deprecation notice, I removed it because of the TODOs already there in relation to supporting windows filepaths. The simple solution seemed to be to move the responsibility for path correctness to the caller, and from there it just followed that if the paths were being moved, might as well move the filesystem access as well. But yeah deprecation would certainly be less jarring for anyone wishing to upgrade.

pdspicer commented:
Alright, `addFile()` is back, deprecation warning included in-process and documentation with reference to the `addCert()` method as a replacement. Added deprecation test as well, and modified build process to fail if tests are not passing.

pdspicer commented:
Let me know any further comments you have on changes, hopefully the latest addresses your questions.

coolaj86 commented:
Thanks so much.

I get really annoyed when I'm trying to help with a project and the maintainer has a bunch of nits. I hope that my nits were valuable and not too nit-picky. If they were, I apologize.

What's your npm name so I can add you as a publisher?

Also, we've moved to https://git.daplie.com/Daplie/node-ssl-root-cas

coolaj86 commented:
bump. Would you like me to add you as a publisher?

pdspicer commented:
Hey sorry, vacationing to a destination with no internet access,  finally back online! Feedback was appreciated, not at all not picky. Would appreciated an add as publisher, npm name is pdspicer.

coolaj86 commented:
Hey @pdspicer the repo is now https://git.coolaj86.com/coolaj86/ssl-root-cas.js and I'd still be happy to add you as a collab.
